src/Infiniband.cc

```diff
@@ -639,11 +639,16 @@ Infiniband::QueuePair::plumb(QueuePairTuple *qpt)
     qpa.rq_psn = qpt->getPsn();
     qpa.max_dest_rd_atomic = 1;
     qpa.min_rnr_timer = 12;
-    qpa.ah_attr.is_global = 0;
+    // NOTE: QueuePairTuple is only used by InfRcTransport. If a different transport
+    // or driver uses QueuePairTuple, we might need a toggle for is_global
+    qpa.ah_attr.is_global = 1;
+    qpa.ah_attr.grh.sgid_index = qpt->getGidIndex();
```

Comment:
I think this is a bug.

Isn't the queue pair tuple `qpt` referring to the remote machine?  But you are setting the source gid index here aren't you.

Reply:
Yep, this is source gid index. If I understand this doc correctly:

https://www.rdmamojo.com/2013/01/12/ibv_modify_qp/

However, 6 lines below is dgid, which is destination gid, and that one I'm setting wrong at moment. Will look into correcting.

Reply:
Per offline conversation, you were right & I was wrong. Making the correction.